numba/targets/arrayobj.py

```diff
@@ -1739,6 +1783,307 @@ def _increment_indices_array(context, builder, arrty, arr, indices, end_flag=Non
     _increment_indices(context, builder, arrty.ndim, shape, indices, end_flag)
 
 
+def make_nditer_cls(nditerty):
```

Comment:
Class doesn't really mean anything, its not clear what it holds (the docstring does explain it's a struct though).